libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/BotFrameworkConfigurationBuilder.cs

```diff
@@ -27,8 +26,19 @@ public BotFrameworkConfigurationBuilder(BotFrameworkOptions botFrameworkOptions)
         /// <seealso cref="ICredentialProvider" />
         public BotFrameworkConfigurationBuilder UseCredentialProvider(ICredentialProvider credentialProvider)
         {
-            _options.CredentialProvider = credentialProvider;
+            BotFrameworkOptions.CredentialProvider = credentialProvider;
+            return this;
+        }
 
+        /// <summary>
+        /// Adds an Error Handler the bot.
```

Comment:
I do not think you intended to commit this with the name (and documentation) of `UseMiddleware`. Seems like it should be `UseErrorHandler`, right?